### ISSUE TITLE
blow-up in the number of samples in 2D orthoplane predictions

### ISSUE THREAD

**Issue by eminorhan** [janelia-cellmap/cellmap-segmentation-challenge#174]
I'm trying to generate predictions from a 2D model for all test crops, following the `predict_2D.py` script under `examples` (except that `crops="test"` instead of `crops="19"`). The predictions for the first crop in `test_crop_manifest.csv` (crop `557`) are completed within a reasonable amount of time, however for the second crop (crop `980`), I noticed that the `dataset_writer` [here](https://github.com/janelia-cellmap/cellmap-segmentation-challenge/blob/d604104ff95a149d3f22f4b0c97e3fdd2dff1e75/src/cellmap_segmentation_challenge/predict.py#L122C1-L122C43) seems to generate a massive number of samples even for a single axis, for example ~165M samples for the first axis, `z` (for comparison, for the first test crop, the corresponding number is only 129 samples). This means that it would take an infeasible amount of time to calculate the predictions even for a single axis (even with a large batch size). I'm just trying to understand why this blow-up seems to happen for the second test crop and how I might prevent it.

**Comment by eminorhan** [comment]
I've been using `2025.9.9.196.dev3+g14815fe23`.

I actually fixed this by passing `dataset_writer.blocks` to a standard pytorch `DataLoader` object, e.g. like [here](https://github.com/eminorhan/cellmap-segmentation-challenge/blob/main/src/cellmap_segmentation_challenge/predict_fsdp.py#L322):
```python
    dataloader = DataLoader(
        dataset_writer.blocks, 
        batch_size=batch_size, 
        sampler=sampler, 
        num_workers=0, 
        pin_memory=True
    )
```
